### PR TITLE
Make name configurable for each health check

### PR DESCRIPTION
**Pull request by karl-sjogren**
Added a wrapper class around IHealthCheck so a name can be supplied. This is really useful when using the endpoint with https://www.nuget.org/packages/AspNetCore.HealthChecks.UI/.

This change makes the service work with this wrapper class internally though it will fall back to the health check class name if no name is specified. I've kept all the extensions using IHealthCheck as well to be backwards compatible.

### CODE REVIEW

File: samples/MvcSample/Startup.cs [hunk 1]
```diff
@@ -20,6 +20,13 @@ public void Configuration(IAppBuilder app)
                 //    @"Data Source=(LocalDB)\v13.0;Integrated Security=True;Initial Catalog=master",
                 //    "select 'a'"),
                 new PingHealthCheck(new PingHealthCheckOptions().AddHost("localhost", 1000)));
+
+            /* Sample with named checks
+            app.UseHealthChecks(
+                "/_health",
```
Comment (kendaleiv) [inline samples/MvcSample/Startup.cs]: Could this be uncommented to make both code paths accessible if this was changed to

```suggestion
                "/_health_named",
```

?

Comment (karl-sjogren) [reply samples/MvcSample/Startup.cs]: Sure, I'll update with that.

I didn't update the logs to use the assigned names (though I think I'll fix that as well now that you mention it). The main usage for this is for usage with the Health Check UI package. Without this addition it looks like this.

image

With it (with different names for checks so I can have multiple service or disk-checks as well) it looks like this.

image

**Comment by kendaleiv** [comment]
Want to update https://github.com/ritterim/RimDev.AspNet.Diagnostics.HealthChecks/blob/72454df974df2d400decc1c053793de72c9936e1/src/RimDev.AspNet.Diagnostics.HealthChecks/Properties/AssemblyInfo.cs#L13-L14 to `1.1.0.0` for both of the versions? I can alternatively do this after merge if you'd prefer but we'll get this published ASAP for you.

**Comment by karl-sjogren** [comment]
Done!

**Comment by kendaleiv** [comment]
Thanks for the contribution! It's published as https://www.nuget.org/packages/RimDev.AspNet.Diagnostics.HealthChecks/1.1.0.

Nice to know this package works with the UI, is there anything special you have to do to get the UI to work with ASP.NET full framework?

**Comment by karl-sjogren** [comment]
Great, thanks!

To get it working with the UI I had to extract the UIResponseWriter from the AspNetCore.HealthChecks.UI.Client package and change HttpContext to IOwinContext.

I could add this in a new PR if you want but it needs a reference to Newtonsoft.Json to work which might cause conflicts with existing projects using other versions of Newtonsoft.Json.

**Comment by kendaleiv** [comment]
I'd accept a PR to add the UI functionality it you'd like but don't feel obligated. I think using an older version of Newtonsoft.Json might be OK. Using an older version might hopefully limit the need for package users to upgrade that dependency.

**Comment by karl-sjogren** [comment]
PRs are what keep open source thriving. Added a PR for the new response writer as #5.

**Pull request merged**